bench/wtperf/wtperf.c

```diff
@@ -548,6 +572,40 @@ worker(void *arg)
 			if ((ret = cursor->insert(cursor)) == 0)
 				break;
 			goto op_err;
+		case WORKER_TRUNCATE:
+			if ((ret = cursor->reset(cursor)) != 0) {
+				lprintf(cfg, ret, 0, "Cursor reset failed");
+				goto err;
+			}
+
+			while ((ret = cursor->next(cursor)) == 0) {
```

Comment:
This loop is going to perform really horribly. We want to match the semantic of oplog as closely as we can. Can you investigate how the oplog works as at https://mongodbcr.appspot.com/6570003/

Then write up a little something about how we could map that onto this truncate implementation in wtperf?